src/Rules/Validation/ScopeValidationMethods.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+
+namespace Hihaho\PhpstanRules\Rules\Validation;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Stringable;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\ObjectType;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionIntersectionType;
+use ReflectionMethod;
+use ReflectionNamedType;
+use ReflectionParameter;
+use ReflectionUnionType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\MethodCall>
+ */
+abstract class ScopeValidationMethods implements Rule
+{
+    abstract public function getNodeType(): string;
+
+    abstract public function processNode(Node $node, Scope $scope): array;
+
+    /**
+     * @phpstan-return ReflectionMethod[]
+     * @throws ReflectionException
+     */
+    protected function getClassMethods(Scope $scope): array
+    {
+        $type = new ObjectType(className: $scope->getClassReflection()?->getName(), classReflection: $scope->getClassReflection());
+        /** @var ReflectionMethod[] $methods */
+        $methods = array_map(static function (string $className): array {
+            return (new ReflectionClass($className))->getMethods();
+        }, $type->getObjectClassNames());
+
+        return $methods;
+    }
+
+    protected function getMethodParameterClassnames(array $method): Collection
```

Comment:
this method receives a list of methods, for which it will get the parameters and return the type.